linux/src/threadpool_linux.c

```diff
@@ -955,33 +857,21 @@ void threadpool_destroy_work_item(THANDLE(THREADPOOL) threadpool, THREADPOOL_WOR
     else
     {
         THREADPOOL* threadpool_ptr = THANDLE_GET_T(THREADPOOL)(threadpool);
-
-        /* Codes_SRS_THREADPOOL_LINUX_05_031: [ threadpool_destroy_work_item shall call sm_exec_begin ]*/
-        SM_RESULT sm_result = sm_exec_begin(threadpool_ptr->sm);
-        if (sm_result != SM_EXEC_GRANTED)
+        
+        /* Codes_SRS_THREADPOOL_LINUX_05_033: [ threadpool_destroy_work_item shall wait for all pending work items to finish execution. ]*/
+        if (InterlockedHL_WaitForValue(&threadpool_work_item->pending_work_item_count, 0, UINT32_MAX) == INTERLOCKED_HL_OK)
         {
-            /* Codes_SRS_THREADPOOL_LINUX_05_032: [ If sm_exec_begin returns SM_EXEC_REFUSED, threadpool_destroy_work_item shall fail. ]*/
-            LogError("sm_exec_begin failed : %" PRI_MU_ENUM, MU_ENUM_VALUE(SM_RESULT, sm_result));
+            /* Codes_SRS_THREADPOOL_LINUX_05_034: [ When all pending work items finish execution, threadpool_destroy_work_item shall acquire the SRW lock in shared mode by calling srw_lock_acquire_shared. ]*/
+            srw_lock_acquire_shared(threadpool_ptr->srw_lock);
```

Comment:
>srw_lock_acquire_shared(threadpool_ptr->srw_lock); [](http://example.com/codeflow?start=12&length=50)

Having a lock over a free does nothing, so let's remove this acquire/release

Reply:
Fixed.